atomic_reactor/plugins/pre_bump_release.py

```diff
@@ -0,0 +1,263 @@
+                        words[1] = next_release
+                        n = splits.index(token)
+                        splits[n] = "=".join(map(quote, words))
+
+                        # Now reconstruct the line
+                        content = " ".join([cmd] + splits) + '\n'
+                        startline = candidate['startline']
+                        endline = candidate['endline']
+                        break
+
+        # We know the label we're looking for is there
+        assert content and startline and endline
+
+        # Re-write the Dockerfile
+        lines = parser.lines
+        del lines[startline:endline + 1]
+        lines.insert(startline, content)
+        parser.lines = lines
+
+    @staticmethod
+    def get_next_release(current_release):
+        try:
+            return str(int(current_release) + 1)
+        except ValueError:
+            isdigit = type(current_release).isdigit
+            first_nondigit = [isdigit(x) for x in current_release].index(False)
+            n = str(int(current_release[:first_nondigit]) + 1)
+            return n + current_release[first_nondigit:]
+
+    def bump(self, repo, branch):
+        # Look in the git repository
+        remote = repo.remotes['origin']
+        if self.push_url:
+            repo.config['push.default'] = 'simple'
+            remote.push_url = self.push_url
+            remote.save()
+
+        # Bump the Release label
+        label_key = 'Release'
+        df_path = self.workflow.builder.df_path
+        parser = DockerfileParser(df_path)
+        current_release = parser.labels[label_key]
+        next_release = self.get_next_release(current_release)
+        self.log.info("New Release: %s", next_release)
+        self.bump_local_file(parser, label_key, next_release)
+
+        # Stage it
+        index = repo.index
+        index.add(os.path.basename(df_path))
+
+        # Commit the change
+        author = Signature(self.author_name, self.author_email)
+        committer = Signature(self.committer_name, self.committer_email)
+        repo.create_commit(branch.name, author, committer, self.commit_message,
+                           index.write_tree(), [repo.head.peel().hex])
+
+        # Push it
+        self.log.info("Pushing to git repository")
+        ssh_command = '/usr/bin/ssh -o StrictHostKeyChecking=no'
+        os.environ['GIT_SSH_COMMAND'] = ssh_command
+
+        # Why doesn't this work?
+        #   remote.push([branch.name])
+        # Instead, run the git command to do it
+        cmd = ['/usr/bin/git', 'push', 'origin']
+        try:
+            output = subprocess.check_output(cmd, stderr=subprocess.STDOUT,
+                                             cwd=repo.workdir)
+            self.log.debug("git (success): %s", output)
+        except subprocess.CalledProcessError as exc:
+            self.log.error("git (%d): %s", exc.returncode, exc.output)
+            raise
+        except AttributeError:
+            # py2.6
```

Comment:
So why not just use the code that is in the except clause for all Python versions? IMO it works on all Python versions we care about, so I don't see the need for the try clause above.